### ISSUE TITLE
Flanking region of genes & METRO P-value Inflation

### ISSUE THREAD

**Issue by Xinyu-Sun** [zhengli09/METRO#4]
Hello,

Thank you for developing this package!  It has been a valuable resource, and I am interested in applying it to our binary trait analysis. However, I have encountered an issue regarding the inflation of P-values. Specifically, when utilizing our NHW GWAS (summary stats), the TWAS lambda yielded a value of about 2.9, with the analysis conducted on AA (~200) and NHW (~200) whole blood eQTL datasets.

To further investigate, I attempted to replicate the findings of the GLGC's HDL GWAS within the NHW population using our eQTL datasets. I observed a similar inflation issue when using a ±1MB flanking region for each gene model. However, when I narrow down the flanking region to ±100kb around each gene, the TWAS lambda aligned more closely with the values reported in the original paper. Moreover, I obtained comparable P-values for PLTP and PPARG.

I would like to know if this inflation issue is a known or expected behavior in the current implementation of the package? 

Thank you!

Xinyu

**Comment by zhengli09** [comment]
Hi Xinyu,

Thank you for your interest in METRO. TWAS is known to suffer from the [LD-hitchhiking effects](https://www.sciencedirect.com/science/article/pii/S0002929722001537) where signals tend to be clustered on the genome. Therefore, if the flanking region size is chosen to be too large, many genes would appear to have small p-values if they are located near the potential biologically relevant genes.  In METRO paper, we set the flanking region size to be 100kb.

Best,
Zheng

**Comment by Xinyu-Sun** [comment]
Thank you for the feedback! I totally agree with the LD effects on clusters of genes near the true causal genes. 

I am sorry that I made a mistake in the previous post:
> Specifically, when utilizing our NHW GWAS (summary stats), the TWAS lambda yielded a value of about 2.9

I meant to say the inflation factor lambda is about 2.9, when I apply ±1MB flanking region on the HDL GWAS. Indeed the inflation problem is gone, when I set the region to 100kb. 

In our project, we tried to apply the method on [AD GWAS](https://www.nature.com/articles/s41588-019-0358-2). In their studies, they phased and imputed the datasets. I noticed the total SNVs in summary stats is roughly 5 times more than that of GLGC's HDL GWAS. Even if I set the flanking region to 100kb, I still see an abnormal P-value issue, which about 2/3 of genes are significant, and many of the P-values are zeros. I also observed the number of SNPs in the gene region is about 3 times more than the HDL GWAS, which is not imputed. 

I am wondering if the problem is due to the number of cis-SNPs in each gene region. The imputation procedure is likely to generate more correlated eQTL weights, and GWAS effects. I would appreciate any suggestions you might have

Thanks,
Xinyu

**Comment by zhengli09** [comment]
In our experience, the density of SNPs in the gene region did not appear to inflate the p-values. We applied METRO to the AAAGC BMI data which contains about five times as many SNPs as the lipid GWAS and the resulting p-values were normal. Genes tend to be detected as TWAS signals if their regions contain GWAS and eQTL signals, or SNPs with relatively large effect sizes. It might be helpful to check GWAS and eQTL effects in those detected gene regions to gain more understanding. Hope these can help.

Best,
Zheng

**Comment by Xinyu-Sun** [comment]
I looked into some of the probamatic (P-value = 0) TWAS results' GWAS and eQTL locuszoom plots, using the Kunkle's 2019 GWAS summary statistics. Here I can show one example. 
![image](https://github.com/zhengli09/METRO/assets/112660039/84e5bc35-5b12-4e71-b3df-2e1dbec74043)


I tried on another more recent AD GWAS, and did not observe the wierd result again. In addition, the similar plot for most significant TWAS signals looked reasonable in this case. Therefore, I think it is likely the problem of the summary statistics, and LD reference.
